### ISSUE TITLE
feat-req: JSON schema generator/validation abstraction

### ISSUE THREAD

**Issue by skwasjer** [confluentinc/confluent-kafka-dotnet#1382]
Description
===========

In relation to my previous PR #1373 I would argue perhaps an improvement would be to provide an abstraction for JSON schema generation and validation, so as to allow other implementations to be used. I am thinking of use cases where you may want to:

- Use [JSON.NET schema](https://www.newtonsoft.com/jsonschema) instead
- Use schemas from other sources (ie. disk, database, or otherwise) as opposed to in-mem generation, loaded through f.ex. `JsonSchema.FromJsonAsync(..)`

Hope it makes some sense, would appreciate your opinion and I will gladly offer my help if so desired.

**Comment by mhowlett** [comment]
yes, certainly open to all that. what's there is an initial implementation.
hopefully the current implementation is not going to be too problematic to extend.
